src/dataflow/src/source/timestamp.rs

```diff
@@ -165,33 +280,71 @@ impl TimestampBindingBox {
         }
     }
 
-    fn add_binding(&mut self, partition: PartitionId, timestamp: Timestamp, offset: MzOffset) {
-        let partition = self
-            .partitions
-            .entry(partition)
-            .or_insert_with(PartitionTimestamps::new);
-        partition.add_binding(timestamp, offset);
+    fn add_partition(&mut self, partition: PartitionId) {
+        if self.partitions.contains_key(&partition) {
+            debug!("already inserted partition {:?}, ignoring", partition);
+            return;
+        }
```

Comment:
so this is intentionally a debug! log instead of `error!` because we can end up adding a partition more than once for some topics -- iirc, we double count a partition because the RT timestamp "discovers" partition 0 and tells us to add it again, even though we start with it.

Reply:
I'll put in a todo to change the timestamper behavior and update the logging and error handling